sdks/go/test/integration/primitives/timers.go

```diff
@@ -0,0 +1,131 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package primitives
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/state"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/timers"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/register"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/passert"
+)
+
+// Based on https://github.com/apache/beam/blob/master/runners/flink/src/test/java/org/apache/beam/runners/flink/PortableTimersExecutionTest.java
+
+func init() {
+	register.DoFn2x0[[]byte, func(string, int)](&inputFn[string, int]{})
+	register.DoFn6x0[beam.Window, state.Provider, timers.Provider, string, int, func(kv[string, int])](&eventTimeFn{})
+	register.Emitter2[string, int]()
+	register.Emitter1[kv[string, int]]()
+}
+
+type kv[K, V any] struct {
+	Key   K
+	Value V
+}
+
+func kvfn[K, V any](k K, v V) kv[K, V] {
+	return kv[K, V]{k, v}
+}
+
+type inputFn[K, V any] struct {
+	Inputs []kv[K, V]
+}
+
+func (fn *inputFn[K, V]) ProcessElement(_ []byte, emit func(K, V)) {
+	for _, in := range fn.Inputs {
+		emit(in.Key, in.Value)
+	}
+}
+
+type eventTimeFn struct {
+	Foo    timers.EventTime
```

Comment:
Foo doesn't seem to be a descriptive name here

Reply:
Agreed. The weird names are from the test I copied it from. The irrelevance of the names it just more forward in Go. Updating those to better names.

Reply:
EventTime is repeating the type, is confusable with the eventTime timestamp that can be passed in, and doesn't say what it's for.

Hence I used "Callback", since it's job is to trigger a callback.